### ISSUE TITLE
Provide support for "? IS NULL" constructs in FB 2.5 [JDBC164]

### ISSUE THREAD

**Issue by firebird-automations** [FirebirdSQL/jaybird#208]
Submitted by: Roman Rokytskyy (rrokytskyy)

Assigned to: Roman Rokytskyy (rrokytskyy)

To resolve the first problem, the request must supply two parameters \(for our Delphi example\):
  WHERE col1 = :param1 OR :param2 IS NULL
  
\- If "param1" is not NULL, the language interface is required to assign the correct value for the first parameter, set the XSQLVAR\.sqlind to NOT NULL and leave XSQLVAR\.sqldata NULL\.

\- If "param2" is NULL, the language interface is required to set the XSQLVAR\.sqlind of both parameters to NULL and leave the XSQLVAR\.sqldata NULL\.

In other words, for the parameter \( ? \) in ? IS NULL:

\- XSQLVAR\.sqlind should be set in accordance with NULL/NON\-NULL state of the parameter\. This is the type of parameter that is described by the new constant SQL\_NULL\.

\- The XSQLVAR\.sqldata of a SQL\_NULL type of parameter should always be passed by the client application as a NULL pointer and should never be accessed\.

Commits: FirebirdSQL/jaybird@747a1275d66563814bd2d3b0e6d54834dec23aff
